dev_guide/secrets.adoc

```diff
@@ -80,6 +108,30 @@ $ oc create -f <filename>
 ----
 ====
 
+[[types-of-secrets]]
+=== Types of Secrets
+
+The value in the `type` field is a hint at the structure of the secret's key names and values.
+
+Some types trigger minimal server-side validation to ensure the presence of specific key names in the secret data:
+
+* `kubernetes.io/service-account-token`. Uses a xref:../admin_guide/service_accounts.adoc#enabling-service-account-authentication[service account token].
+* `kubernetes.io/dockercfg`. Uses the xref:../dev_guide/managing_images.adoc#allowing-pods-to-reference-images-from-other-secured-registries[*_.dockercfg_* file] 
+for required Docker credentials.
+* `kubernetes.io/dockerconfigjson`. Uses the xref:../dev_guide/builds/build_inputs.adoc#using-docker-credentials-for-private-registries[*_.docker/config.json_* file] 
+for required Docker credentials. 
+* `kubernetes.io/basic-auth`. Use with xref:../dev_guide/builds/build_inputs.adoc#source-secrets-basic-authentication[Basic Authentication].
+* `kubernetes.io/ssh-auth`. Use with xref:../dev_guide/builds/build_inputs.adoc#source-secrets-ssh-key-authentication[SSH Key Authentication].
+* `kubernetes.io/tls`. Use with xref:../dev_guide/builds/build_inputs.adoc#source-secrets-trusted-certificate-authorities[TLS certificate authorities]
+
+You can specify other arbitrary types (such as `example.com/my-secret-type`), which are not enforced server-side, 
+but indicate that the creator of the secret intended to conform to the key/value requirements of that type.
```

Comment:
The rest of this section confuses me.
- If opaque is default, why do I need to specify it? (132)
- If opaque is default, what key/value requirements of a type are intended? (128)
- Are users going to apply "opaque" to one of the types that is usually validated?

Could we replace lines 128-133 with something like "Arbitrary secrets are `opaque` by default. Opaque secrets do not claim to conform to any convention for key names or values."